packages/coil-anonymous-tokens/src/index.ts

```diff
@@ -0,0 +1,151 @@
+import { randomBytes } from 'crypto'
+
+import { BigInteger } from 'jsbn'
+import {
+  longHash,
+  hexString,
+  hashAndBlindMessage,
+  unblindSignature,
+  verifySignature,
+  PublicRSAKey
+} from 'blind-signature'
+
+export function base64url(buf: Buffer) {
+  return buf
+    .toString('base64')
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=/g, '')
+}
+
+export const TOKEN_PREFIX = 'anonymous_token:'
+export interface SignedToken {
+  value: string
+  month: string
+  signature: hexString
+}
+
+export interface TimestampedSignature {
+  signature: string
+  month: string
+}
+
+export interface PublicFields extends PublicRSAKey {
+  month: string
+}
+
+// TODO: should these be allowed to be async?
+// TODO: went for localforage-like, could chang
+export interface TokenStore {
+  getItem: (key: string) => Promise<string>
+  setItem: (key: string, value: string) => Promise<string>
+  removeItem: (key: string) => Promise<void>
+  iterate: (
+    fn: (
+      value: string,
+      key: string,
+      iterationNumber: number
+    ) => SignedToken | undefined
+  ) => Promise<SignedToken | undefined>
+}
+
+export class AnonymousTokens {
+  private store: TokenStore
```

Comment:
This can use constructor assignment

```suggestion
  constructor(private store: TokenStore) {
  }
```